src/fabric/src/fabric_view.erl

```diff
@@ -187,23 +187,33 @@ possibly_embed_doc(
     #view_row{id = reduced} = Row
 ) ->
     Row;
+possibly_embed_doc(
+    _State,
+    #{id := reduced} = Row
+) ->
+    Row;
 possibly_embed_doc(
     _State,
     #view_row{value = undefined} = Row
 ) ->
     Row;
```

Comment:
I think this was originally added here to implement a shortcut.  But since keys with `undefined` value are not stored in the map representation, the same cannot be realized in that case so it will go through the code path as you suggested.  I do not think this line should be removed for now but along with retiring the `#view_row{}` record.